### PR TITLE
storage/zynq7000: add _storage_devCtx_t structure

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

PD-216

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (zynq7000).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (https://github.com/phoenix-rtos/phoenix-rtos-corelibs/pull/13).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/zynq7000-flash/flashdrv.c [hunk 1]
```diff
@@ -674,6 +674,7 @@ int flashdrv_devInit(storage_t *strg)
 
 	/* Initialize dev structure for new region */
 	strg->dev = malloc(sizeof(storage_dev_t));
+	strg->dev->ctx = malloc(sizeof(storage_devCtx_t));
```
Comment (agkaminski) [inline storage/zynq7000-flash/flashdrv.c:677]: No check for this malloc

File: storage/zynq7000-flash/flashdrv.c [hunk 2]
```diff
@@ -551,13 +551,13 @@ static ssize_t flashdrv_mtdWrite(struct _storage_t *strg, off_t offs, const void
 	ssize_t res = 0;
 	size_t tempSz = 0, chunkSz = 0;
 
-	mutexLock(fdrv_common.regs[strg->dev->id].lock);
+	mutexLock(fdrv_common.regs[strg->dev->ctx->id].lock);
 	chunkSz = offs % strg->dev->mtd->writeBuffsz;
 	if (chunkSz != 0) {
 		chunkSz = len < (strg->dev->mtd->writeBuffsz - chunkSz) ? len : (strg->dev->mtd->writeBuffsz - chunkSz);
-		res = _flashdrv_pageProgram(strg->dev->id, strg->start + offs, data, chunkSz);
+		res = _flashdrv_pageProgram(strg->dev->ctx->id, offs, data, chunkSz);
```
Comment (agkaminski) [inline storage/zynq7000-flash/flashdrv.c:558]: Removed `strg->start` without corresponding a change in `_flashdrv_pageProgram`. Is this intended?

File: storage/zynq7000-flash/flashdrv.c [hunk 3]
```diff
@@ -566,15 +566,15 @@ static ssize_t flashdrv_mtdWrite(struct _storage_t *strg, off_t offs, const void
 
 	while (tempSz < len) {
 		chunkSz = ((len - tempSz) > strg->dev->mtd->writeBuffsz) ? strg->dev->mtd->writeBuffsz : (len - tempSz);
-		res = _flashdrv_pageProgram(strg->dev->id, strg->start + offs + tempSz, (const char *)data + tempSz, chunkSz);
+		res = _flashdrv_pageProgram(strg->dev->ctx->id, offs + tempSz, (const char *)data + tempSz, chunkSz);
```
Comment (agkaminski) [inline storage/zynq7000-flash/flashdrv.c:569]: Removed `strg->start` without corresponding a change in `_flashdrv_pageProgram`. Is this intended?

File: storage/zynq7000-flash/flashdrv.c [hunk 4]
```diff
@@ -614,10 +614,10 @@ int flashdrv_done(storage_t *strg)
 
 	res = flashdrv_blkSync(strg);
 
-	free(fdrv_common.regs[strg->dev->id].buff);
-	fdrv_common.regs[strg->dev->id].buff = NULL;
+	free(fdrv_common.regs[strg->dev->ctx->id].buff);
+	fdrv_common.regs[strg->dev->ctx->id].buff = NULL;
 
-	resourceDestroy(fdrv_common.regs[strg->dev->id].lock);
+	resourceDestroy(fdrv_common.regs[strg->dev->ctx->id].lock);
```
Comment (agkaminski) [inline storage/zynq7000-flash/flashdrv.c:620]: Missing `free(strg->dev->ctx)`

**Review by agkaminski — CHANGES_REQUESTED** [review]

File: storage/zynq7000-flash/flashdrv.c (hunk 2 above)
Comment (HBuczynski) [reply storage/zynq7000-flash/flashdrv.c]: Nope, it was related to other changes not connected to this PR. Thanks.

**Review by agkaminski — APPROVED** [review]

File: storage/zynq7000-flash/flashdrv.c (hunk 1 above)
Comment (HBuczynski) [reply storage/zynq7000-flash/flashdrv.c]: done

**Pull request merged**